lib/service/login.dart

```diff
@@ -32,9 +46,21 @@ class LoginService {
     }
   }
 
-  Future<bool> isLoggedIn() async {
+  Future<bool> isLoggedIn(BuildContext context) async {
+    final accessData =
+        Provider.of<AccessDataChangeNotifier>(context, listen: false);
+    if (accessData.hasData && accessData.isLoggedIn) {
+      return true;
+    }
+
     final prefs = await SharedPreferences.getInstance();
 
+    Provider.of<AccessDataChangeNotifier>(context, listen: false).init(
+      shopUrl: prefs.getString('shopUrl'),
+      accessToken: prefs.getString("access_token"),
```

Comment:
in den prefs sind die bis jetzt in camel case drin `accessToken` und `refreshToken`

Reply:
Außerdem wird das knallen wenn die pref noch nicht gesetzt ist, das macht das letzte return momentan unnötig

Reply:
Warum springt das mit einer Exception raus?

Reply:
Wenn ich direkt darueber `(await SharedPreferences.getInstance()).clear();` ausfuehre bekomme ich keine exception und werde wie gewuenscht zum login geleitet

Reply:
ok dann scheint es zu klappen, sind returnTypes immer nullable in dart? 
weil in dem fall `getString("wasAuchImmer")` ja keinen string sondern wahrscheinlich null zurückliefert oder wird das intern nach `""` gecastet?

Reply:
Genau jeder Typ kann auch `NULL` sein, z.B. wenn du eine variable nur deklarierst und nicht initialisierst ist diese auch erstmal null selbst mit festem typ